CRO i5: Add i5 AB test

elliottprogrammer opened the pull request:
### Changes proposed in this Pull Request

Add new AB test variation for the Jetpack Plans page.

#### Implementation notes:

I pointed the new AB test variation to a new `products-grid-alt-3` folder & component.

Additional note:  The features list in top 3 plans in this new products-grid-alt-3 page are not working correctly on this page. I didn't take the time to repair this because we will be changing these things in upcoming PR's.

### Testing instructions

1. Run this PR
2. Visit the Plans page: `http://calypso.localhost:3000/plans/:site`
3. Verify there is a new `jetpackConversionRateOptimization` AB test option labelled: `i5 - Saas table design`
4. Select the `i5 - Saas table design` variant.
5. Verify you can see the text "`i5 - Saas table design`" (See screenshot) - This means you are on the variant plans grid page (`ProductsGridAlt3`)
![Markup 2020-11-09 at 13 42 10](https://user-images.githubusercontent.com/11078128/98599823-7563a900-2291-11eb-83a1-ae5a86b27905.png)

6. Select the other A/B test variants to make sure they are all operating correctly. 


Fixes # 1196341175636977-as-1199149328812557/f

elliottprogrammer commented on apps/editing-toolkit/editing-toolkit-plugin/premium-content/blocks/buttons/index.js:
```diff
@@ -28,6 +28,15 @@ const settings = {
 		alignWide: false,
 		lightBlockWrapper: true,
 	},
+	attributes: {
+		isPremiumContentChild: {
+			type: 'bool',
+			default: true,
+		},
+	},
+	providesContext: {
+		isPremiumContentChild: 'isPremiumContentChild',
+	},
```
Hmm, what the heck is this?  I have no idea how this got into my PR. I've never touched this file in my life.  ??

monsieur-z replied:
It's really strange, it's the exact same file on master. I thought maybe formatting issue, but the lines are new, not modified. Maybe a rebase of master would fix this.

monsieur-z commented on client/lib/abtest/active-tests.js:
```diff
@@ -197,9 +197,10 @@ export default {
 			'v0 - Offer Reset': 0, // Offer Reset
 			'v1 - 3 cols layout': 0, // first Offer Reset iteration (3 layout columns + simpler cards)
-			'v2 - slide outs': 100, // second Offer Reset iteration (reorder & slide outs)
+			'i2 - slide outs': 100, // second Offer Reset iteration (reorder & slide outs)
```
I suggest we leave the previous iterations as is. It seems safer.

monsieur-z reviewed (changes requested):
It works well! I think we should leave the previous test name as is, though.

Just FYI, I've created a new grid and updated the controller too, in the layout task. Feel free to leave them in the PR, I'll resolve the conflicts when I rebase.

elliottprogrammer commented:
> It works well! I think we should leave the previous test name as is, though.
> 
> Just FYI, I've created a new grid and updated the controller too, in the layout task. Feel free to leave them in the PR, I'll resolve the conflicts when I rebase.

Yes you're right, changing the previous iteration doesn't feel very safe and just makes it more confusing.

Ok I've changed the previous iteration back to "`v2`" and also rebased, which appears to have removed the mystery code from this PR revision.  Oh, sorry about updating the controller, I didn't mean to overlap or create conflicts 🤦🏼‍♂️  Yes, just clear out my overlapping code when you rebase. Apologies again and thanks 👍

monsieur-z commented:
> Ok I've changed the previous iteration back to "v2" and also rebased, which appears to have removed the mystery code from this PR revision.

Nice!

> Oh, sorry about updating the controller, I didn't mean to overlap or create conflicts 🤦🏼‍♂️ Yes, just clear out my overlapping code when you rebase. Apologies again and thanks

No worries! It was easier to test the PR the way you did it. Just wanted to give you a heads-up about how we'll merge our work.
